pysparkling/utils.py

```diff
@@ -329,5 +329,39 @@ def merge_rows(left, right, on=_sentinel):
     ))
 
 
+def strhash(string):
+    """
+    Old python hash function as described in PEP 456, excluding prefix, suffix and mask.
+
+    :param string: string to hash
+    :return: hash
+    """
+    if string == "":
+        return 0
+
+    x = ord(string[0]) << 7
+    for c in string[1:]:
+        x = ((1000003 * x) ^ ord(c)) & (1 << 32)
+    x = (x ^ len(string))
+    return x
+
+
+def tuplehash(tuple):
+    """
+    Old python hash function for tuples.
+
+    :param tuple: tuple to hash
+    :return: hash
+    """
+    if isinstance(tuple, int):
```

Comment:
Could you rename the `tuple` variable as it clashes with a builtin name and apparently can be an int, not only a tuple?

Reply:
yeah